op-service/txmgr/metrics/tx_metrics.go

```diff
@@ -42,10 +63,59 @@ func MakeTxMetrics(ns string, factory metrics.Factory) TxMetrics {
 			Help:      "Latency of a confirmed transaction in milliseconds",
 			Subsystem: "txmgr",
 		}),
+		currentNonce: factory.NewGauge(prometheus.GaugeOpts{
+			Namespace: ns,
+			Name:      "current_nonce",
+			Help:      "Current nonce of the from address",
+			Subsystem: "txmgr",
+		}),
+		txConfirmed: factory.NewCounterVec(prometheus.CounterOpts{
+			Namespace: ns,
+			Name:      "tx_confirmed_count",
+			Help:      "Count of the transactions that have been confirmed. Label is receipt status.",
+			Subsystem: "txmgr",
+		}, []string{"status"}),
+		txPublished: factory.NewCounter(prometheus.CounterOpts{
+			Namespace: ns,
+			Name:      "tx_published_count",
+			Help:      "Count when transactions are submitted",
+			Subsystem: "txmgr",
+		}),
+		txPublishError: factory.NewCounterVec(prometheus.CounterOpts{
+			Namespace: ns,
+			Name:      "tx_publish_error_count",
+			Help:      "Count of publish errors. Labells are sanitized error strings",
+			Subsystem: "txmgr",
+		}, []string{"error"}),
+		lastPublishTime: factory.NewGauge(prometheus.GaugeOpts{
+			Namespace: ns,
+			Name:      "last_publish_time_unix",
+			Help:      "Unix timestamp of the last tx publish",
+			Subsystem: "txmgr",
+		}),
+		lastConfirmTime: factory.NewGauge(prometheus.GaugeOpts{
+			Namespace: ns,
+			Name:      "last_confirm_time_unix",
+			Help:      "Unix timestamp of the last tx confirmation",
+			Subsystem: "txmgr",
+		}),
+		rpcError: factory.NewCounter(prometheus.CounterOpts{
+			Namespace: ns,
+			Name:      "rpc_error_count",
+			Help:      "Temporrary: Count of RPC errors (like timeouts) that have occurrred",
+			Subsystem: "txmgr",
+		}),
 	}
 }
 
-func (t *TxMetrics) RecordL1GasFee(receipt *types.Receipt) {
+func (t *TxMetrics) RecordNonce(nonce uint64) {
+	t.currentNonce.Set(float64(nonce))
+}
+
+// TxConfirmed records lots of information about the confirmed transaction
+func (t *TxMetrics) TxConfirmed(receipt *types.Receipt) {
+	t.lastConfirmTime.SetToCurrentTime()
+	t.txConfirmed.WithLabelValues(receiptStatusString(receipt)).Inc()
```

Comment:
When you switch to `EventVec`, we'd record the timestamp for every individual receipt status, which is probably also a good little improvement.